fdbserver/masterserver.actor.cpp

```diff
@@ -250,6 +250,9 @@ struct MasterData : NonCopyable, ReferenceCounted<MasterData> {
 	// Captures the latest commit version targeted for each storage server in the cluster.
 	VersionVector ssVersionVector;
 
+	// The previous commit versions per tlog
+	std::map<uint16_t, Version> tpcvMap;
```

Comment:
This could be an array (or vector) for faster access. We know the number of tlogs during recruitment.

Reply:
We are considering making `VersionVector` a templated structure so it could be used for both tags and tLogs. e.g. `VersionVector<Tag> ssVersionVector` and `VersionVector<uint16_t>`.. it uses an unordered map hash. If we were to use an array we would allocate it to be the size=#tlogs which may be more than is needed for typical transactions. On balance I think migrating it to the VersionVector structure on the next pass is preferable

Reply:
Agree `VersionVector<uint16_t> tpcvVector` is preferable. I still think the implementation for tlogs' version vector can be an array, because it's fixed size and small ---- a few extra bytes are not a big overhead and saving CPU for master could be a big deal.